File: 4008-restore-finishing-order/restore-finishing-order.py
class Solution:
    def recoverOrder(self, order: List[int], friends: List[int]) -> List[int]:
        output = []
        frndHash = {}
        for each in friends:
            frndHash[each] = 1
        for each in order:
            if each in frndHash:
                output.append(each)
        return output


        

File: 4008-restore-finishing-order/restore-finishing-order.java
class Solution {
    public int[] recoverOrder(int[] order, int[] friends) {
        List<Integer> output = new ArrayList<>();
        Set<Integer> set = new HashSet<>();
        for (int f : friends) {
            set.add(f);
        }
        for (int each : order){
            if (set.contains(each)){
                output.add(each);
            }
        }

        int[] result = new int[output.size()];
        for (int i = 0; i < output.size(); i++) {
            result[i] = output.get(i);
        }
        return result;
        }
}